docs/release-notes/flink-1.11.md

```diff
@@ -0,0 +1,291 @@
+`AsyncWaitOperator` will be allowed to be chained by default with all operators, except of tasks with `SourceFunction`.
+This mostly revert limitation introduced as a bug fix for [FLINK-13063](https://issues.apache.org/jira/browse/FLINK-13063).
+
+#### Changed argument types of ShuffleEnvironment#createInputGates and #createResultPartitionWriters methods ([FLINK-16586](https://issues.apache.org/jira/browse/FLINK-16586))
+The argument type of methods `ShuffleEnvironment#createInputGates` and `#createResultPartitionWriters` are adjusted from `Collection` to `List` for satisfying the order guarantee requirement in unaligned checkpoint.
+It will break the compatibility if users already implemented a custom `ShuffleService` based on `ShuffleServiceFactory` interface.
+
+#### Deprecation of CompositeTypeSerializerSnapshot#isOuterSnapshotCompatible ([FLINK-17520](https://issues.apache.org/jira/browse/FLINK-17520))
+The `boolean isOuterSnapshotCompatible(TypeSerializer)` on the `CompositeTypeSerializerSnapshot` class has been deprecated, in favor of a new `OuterSchemaCompatibility resolveOuterSchemaCompatibility(TypeSerializer)` method.
+Please implement that instead.
+Compared to the old method, the new method allows composite serializers to signal state schema migration based on outer schema and configuration.
+
+#### Removal of deprecated TimestampExtractor ([FLINK-17655](https://issues.apache.org/jira/browse/FLINK-17655))
+The long-deprecated `TimestampExtractor` was removed along with API methods in the DataStream API.
+Please use the new `TimestampAssigner` and `WatermarkStrategies` for working with timestamps and watermarks in the DataStream API.
+
+#### Deprecation of ListCheckpointed interface ([FLINK-6258](https://issues.apache.org/jira/browse/FLINK-6258))
+The `ListCheckpointed` interface has been deprecated because it uses Java Serialization for checkpointing state which is problematic for savepoint compatibility.
+Use the `CheckpointedFunction` interface instead, which gives more control over state serialization.
+
+#### Extended CheckpointListener interface ([FLINK-8871](https://issues.apache.org/jira/browse/FLINK-8871))
+Added a new method `#notifyCheckpointAborted(long checkpointId)` to the `CheckpointListener` interface.
+Existing users which do not want to act on checkpoint aborted notifications should provide empty method to implement it.
+
+#### Removal of deprecated state access methods ([FLINK-17376](https://issues.apache.org/jira/browse/FLINK-17376))
+We removed deprecated state access methods `RuntimeContext#getFoldingState()`, `OperatorStateStore#getSerializableListState()` and `OperatorStateStore#getOperatorState()`.
+This means that some code that was compiled against Flink 1.10 will not work with a Flink 1.11 cluster.
+An example of this is our Kafka connector which internally used `OperatorStateStore.getSerializableListState`.
```

Comment:
Are the alternatives for the state access now @aljoscha?

Reply:
If there are, @aljoscha could you propose a change in a separate PR? I'm about to merge this one.